src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -208,7 +208,7 @@ private async Task<string> ComputeValueAsync(ComplexTargetFramework complexTarge
 
         private static bool IsNetCore5OrHigher(string targetFrameworkAlias)
         {
-            return targetFrameworkAlias.Contains("5.0") || targetFrameworkAlias.Contains("6.0");
+            return targetFrameworkAlias.Contains("5.0") || targetFrameworkAlias.Contains("6.0") || targetFrameworkAlias.Contains("7.0");
```

Comment:
This problem is going to surface again with .NET 8.

We have access to the TargetFrameworkIdentifier and TargetFrameworkVersion here. Can we use those instead?

For example,

```c#
return TargetFrameworkIdentifier == ".NETCoreApp" && ParseTargetFrameworkVersion(TargetFrameworkVersion) > 6.0m;
```

The comparisons should use the correct `StringComparer`. `TargetFrameworkVersion` has form `v6.0`, so `ParseTargetFrameworkVersion` would need to remove the leading `v` and parse the rest using `decimal.TryParse`. If it's possible that the version has form `0.0.0` then we should use `System.Version` instead.

Reply:
Ideally, yes. There's a bigger problem around that is the property querying is for the previous TargetFramework, not the one that is actually being changed to. But I agree, I think it's a good time to rethink how we're handling it. I'll try to come up with something today.